### DISCUSSION TITLE
Is it possible to use cache in sixel-based image preview?

### DISCUSSION THREAD

**Discussion by smartding** [gokcehan/lf#1871] [Q&A] [answered]
The sixel-based image preview in the wiki use the following script. That means chafa is called every time an image is previewed. 

Is is possible to use cache, as in the ueberzug-based preview method.

```bash
#!/bin/sh
case "$(file -Lb --mime-type -- "$1")" in
  image/*)
    chafa -f sixel -s "$2x$3" --animate off --polite on "$1"
    exit 1
    ;;
  text/*)
    cat "$1"
    ;;
esac
```

**Comment by veltza** [comment, marked as answer]
Change `exit 1` to `exit 0` and it will cache the sixels in memory.

**Reply by smartding** [reply]
Any side effect for `exit 0` you're aware of?

**Reply by veltza** [reply]
Only increased memory usage because lf never flushes the cache unless you reload files (ctrl-r) or resize the terminal window.

**Comment by og900aero** [comment]
> The sixel-based image preview in the wiki use the following script. That means chafa is called every time an image is previewed.
> 
> Is is possible to use cache, as in the ueberzug-based preview method.
> 

I use this preview script for sixel:

```
#!/bin/sh

image() {
	geometry="$(($2-2))x$3"
	chafa "$1" -f sixel -s "$geometry" --animate false
}

CACHE="$HOME/.cache/lf/thumbnail.$(stat --printf '%n\0%i\0%F\0%s\0%W\0%Y' -- "$(readlink -f "$1")" | sha256sum | awk '{print $1}'))"

case "$(file -Lb --mime-type -- "$1")" in
  image/*)
      image "$1" "$2" "$3" "$4" "$5"
 ;;
 */pdf)
   [ ! -f "${CACHE}.jpg" ] && \
		pdftoppm -jpeg -f 1 -singlefile "$1" "$CACHE"
   image "${CACHE}.jpg" "$2" "$3" "$4" "$5"
   ;;
  video/*)
  [ ! -f "${CACHE}.jpg" ] && \
	ffmpegthumbnailer -i "$1" -o "${CACHE}.jpg" -s 0 -q 5
  image "${CACHE}.jpg" "$2" "$3" "$4" "$5"
  ;;
  *)
      file -ibL "$1" | grep -q text && batorcat "$1" || file -Lb "$1"
   ;;
esac
```

**Reply by smartding** [reply]
chafa is still called each time for images.

**Reply by og900aero** [reply]
> chafa is still called each time for images.

So?
